Feat/order update callback

BennyThadikaran opened the pull request:
This PR adds an optional callback function that can be attached to the `orderupdate.OrderSocket` class.

- If callback is attached, execute the function passing the order dict as an argument.
- If not attached, the default functionality is preserved. (i.e. Print the order details)

The callback feature allows users to process the order data and perform various operations like:
- store it to a database
- perform follow up operations like monitor open positions, set stop loss orders if necessary.

The order websocket feature is great because it avoids the need to setup webhook urls and instead get the order updates locally.

The `Live Order Update Usage` section of README has been updated showing the usage of the callback feature.

Do let me know if any additional changes are required.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Introduced support for custom callback functions to handle order updates, allowing users to define their own logic for processing order update events.

- **Documentation**
  - Updated usage examples to demonstrate how to use the new order update callback functionality.

- **Tests**
  - Added unit tests to verify both default and custom callback behaviors for order updates.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

BennyThadikaran commented:
I have written tests for `OrderSocket.handle_order_update` using unittest module. I havent seen any test files, so not included it. If you prefer another framework let me know.

```py
class TestHandleOrderUpdate(unittest.TestCase):
    def setUp(self) -> None:
        self.order_cls = OrderSocket()

    @patch("builtins.print")
    def test_order_update_with_no_callback(self, mock_print: Mock):
        order_data = {
            "Type": "order_alert",
            "Data": dict(orderNo=123, status="order_status"),
        }

        # With order No
        self.order_cls.handle_order_update(order_data)

        # No order No
        order_data["Data"].pop("orderNo")
        self.order_cls.handle_order_update(order_data)

        # Unknown order type
        order_data["Type"] = "unknown"
        self.order_cls.handle_order_update(order_data)

        self.assertEqual(mock_print.call_count, 3)

    def test_order_update_with_callback(self):
        def on_order_update(_):
            pass

        order_data = dict(Type="order_alert")

        mock_cb = Mock(spec=on_order_update)

        self.order_cls.on_update = mock_cb

        self.order_cls.handle_order_update(order_data)

        mock_cb.assert_called_once_with(order_data)


if __name__ == "__main__":
    unittest.main()
```

dhan-oss commented:
Hello @BennyThadikaran 

First of all, great to have your first PR. We have made some major changes to the repository and it will be great if you can fork the latest update and then create the PR.

BennyThadikaran commented:
@dhan-oss
I have pulled the latest changes. The only additional change from my previous PR:

- Removed the async-await keyword for `OrderUpdate.handle_order_update` as there is no awaiting inside the function.
- Added unittest using pytest. Tests are passing except for code coverage failure.

dhan-oss commented:
#57

dhan-oss commented:
FIx #57
